C-Sharp.md

```diff
@@ -1,6 +1,232 @@
+
+Alternatively, you could also download [Visual Studio Community 2015]  [visualstudio]  which is an extensible [IDE][ide] used by most professionals for creating enterprise applications.
+
+###Your First C# Program
+
+```csharp
+//this is the single line comment
+
+/** This is multiline comment,
+compiler ignores any code inside comment blocks.
+**/
+
+//This is the namespace, part of the standard .NET Framework Class Library
+using System;
+// namespace defines the scope of related objects into packages
+namespace Learning.CSharp
+{  
+  // name of the class, should be same as of .cs file
+  public class Program
+  {
+    //entry point method for console applications
+   public static void Main()
+    {
+      //print lines on console
+      Console.WriteLine("Hello, World!");
+      //Reads the next line of characters from the standard input stream.Most common use is to pause program execution before clearing the console.
+      Console.ReadLine();
+    }
+  }
+}
+```
+If you are using visual Studio, follow [this](https://msdn.microsoft.com/en-us/library/k1sx6ed2.aspx) to create console application and understand your first C# program.
+
+###Types and variables
+C# is a strongly typed language. Every variable has a type.
+Every expression or statement evaluates to a value.
+There are two kinds of types in C#
+  * Value types
+  * Reference types.
+
+*_Value Types_* : Variables that are value types directly contain values. Assigning one value type variable to another copies the contained value.
+
+```chsarp
+int a = 10;
+int b = 20;
+a=b;
+Console.WriteLine(a); //prints 20
+Console.WriteLine(b); //prints 20
+```
+All these data types are also wrapped in classes in the `System namespace`. For instance
+`int` is the short name for `System.Int32` and `double` is the short form of `System.Double`.
+
+
+Few examples of built-in data types are `int, float, double, decimal, char and string`.
+
+Type | Example | Description
+--- | --- | ---
+*Integer* | `int fooInt = 7;` | **Signed 32-bit** Interger
+*Long* | `long fooLong = 3000L;` | **Signed 64-bit** integer.**L is used to specify that this variable value is of type long/ulong**
+*Double* | `double fooDouble = 20.99;` | Precision: **15-16 digits**
+*Float* | `float fooFloat = 314.5f;` | Precision: **7 digits**.**F is used to specify that this variable value is of type float**
+*Decimal* | `decimal fooDecimal = 23.3m;` | Precision: **28-29 digits**.Its more precision and smaller range makes it appropriate for **financial and monetary calculations**
+*Char* | `char fooChar = 'Z';` | A  single **16-bit Unicode character**
+*Boolean* | `bool fooBoolean = false;` | Boolean - **true & false**
+*String* |`string fooString = "\"escape\" quotes and add \n (new lines) and \t (tabs);`|**A string of Unicode characters.**
+
```

Comment:
check this link [msdn](https://msdn.microsoft.com/en-us/library/c8f5xwh7.aspx) and see the list on the left hand side.